### DISCUSSION TITLE
Query parameters don't get interpreted when using hash mode

### DISCUSSION THREAD

**Discussion by HomoCodens** [plausible/analytics#1967] [Anything Else]
I'm maintaining a Vue.js SPA which uses hash based routing and we would like to add Plausible tracking.

It appears to me that any query parameters after the `#` don't get picked up but simply interpreted as part of the hash fragment.

In the [docs](https://plausible.io/docs/custom-query-params) you state that most query parameters should be stripped out but I'm seeing URLs like `/#/detail?country=de` in my reports.

I also tried UTM tracking by adding a ficticious `?utm_campaign=Test` to a request but nothing shows up under "UTM Campaigns" on the Top Sources card (the request does get recorded though).

Am I doing something wrong? I don't think Vue gives me too much control over where the query parameters get placed in the URL.

**Comment by RobertJoonas** [comment]
Hi @HomoCodens!

I see what the problem is. Basically, if a `#` symbol is present in the URL, then everything that comes after that, is interpreted as the hash part and will be shown in your dashboard (assuming the hash script extension here). A quick example:

1) `https://plausible.io/#hash_part?utm_source=Twitter`
     -> pathname is the whole URL (even query params are interpreted as the hash part)
     -> utm_source is ignored

2) `https://plausible.io/?utm_source=Twitter#hash_part`
     -> pathname is `https://plausible.io/#hash_part` (query params are stripped)
     -> utm_source is Twitter

If a hash is used in combination with UTM tags (or other query params), then according to the generic URI syntax, the hash part (or "fragment") should always come last. So in my two examples, the second one is what you should aim for.

Hope this makes sense, thanks!

**Comment by HomoCodens** [comment]
Hi

Thanks for confirming what I had already suspected.

I had interpreted the hash extension for Plausible to mean compatibility with hash routing on SPAs but it seems I was mistaken there and it simply means inclusion of the hash fragment, whatever it may be.

Unfortunately Vue does not give great control over how and where the query params appear. Also they look more 'natural' at the end of the URL.
Maybe this is a good time to ditch hash routing for me.
